db-client-java/src/test/java/com/eventstore/dbclient/PersistentSubscriptionTestsBase.java

```diff
@@ -11,19 +11,22 @@
     @Rule
     public final EventStoreTestDBContainer server = new EventStoreTestDBContainer(true);
     protected EventStoreDBPersistentSubscriptionsClient client;
+    protected EventStoreDBClient streamClient;
 
     @Before
     public void before() throws Throwable {
         client = server.getPersistentSubscriptionsClient();
+        streamClient = server.getClient();
     }
 
     @After
     public void after() {
-        if(client == null)
+        if(client == null && streamClient == null)
```

Comment:
**CONCERN:** Shouldn't we have separate ifs for that? I'm not sure if that's a real issue, but if `client` is `null`, but `streamClient` is not, then `streamClient` won't be shut down. The same goes the other way round.

Reply:
I don't get your remark as that test makes sure it's only when both are `null` that exit the function. Otherwise, we shutdown both clients.

Reply:
My point was that if only one is null then it won't be shut down. That's minor.

Reply:
Ah, nvm, I see that there are additional ifs now below in try/catch.